kubetest2/pkg/app/testers/standard/ginkgo/ginkgo.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ginkgo
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"k8s.io/test-infra/kubetest2/pkg/app/testers"
+	"k8s.io/test-infra/kubetest2/pkg/app/testers/standard/ginkgo/kubectl"
+	"k8s.io/test-infra/kubetest2/pkg/exec"
+	"k8s.io/test-infra/kubetest2/pkg/types"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	binary = "ginkgo" // TODO(RonWeber): Actually find these binaries.
+
+)
+
+var (
+	kubeRoot     string
+	platformPath string
+)
+
+const usage = "Eh, we'll figure it out."
+
+func init() {
+	testers.Register("ginkgo", usage, NewTester)
+}
+
+// Tester implements a kubetest2 types.Tester that exec's it's arguments
+type Tester struct {
+	conformanceTest bool
+
+	flakeAttempts string
+	parallel      string
+	host          string
+	skipRegex     string
+
+	deployer types.GinkgoableDeployer
+}
+
+// NewTester creates a new Tester
+func NewTester(common types.Options, testArgs []string, deployer types.Deployer) (types.Tester, error) {
+	wd, _ := os.Getwd()
+	kubeRoot = filepath.Join(wd, "kubernetes")                            //TODO(RonWeber): Real logic to get kubernetes
+	platformPath = filepath.Join(kubeRoot, "platforms", "linux", "amd64") //TODO(RonWeber): Real logic to figure out current platform.
+
+	t := Tester{}
+
+	gd, ok := deployer.(types.GinkgoableDeployer)
+	if !ok {
+		return nil, fmt.Errorf("the ginkgo tester does not support this deployer")
```

Comment:
why can't we default if it's just a Deployer?

Reply:
Because we still need Kubeconfig().

I'm hesitant to rely on the KUBECONFIG environment variable, because in the case where a non-Kubeconfig()-having deployer gets called without --up, there's no point where the deployer has a chance to set environment variables.

Reply:
> I'm hesitant to rely on the KUBECONFIG environment variable, because in the case where a non-Kubeconfig()-having deployer gets called without --up, there's no point at which the deployer has a chance to set environment variables.

It's 100% standard for kubernetes tooling to work with KUBECONFIG or the default path without having one supplied. many deployers (minikube, GKE, ...) will just merge their cluster credentials into the current KUBECONFIG which is usually the default ($HOME/.kube/config on macOS / Linux)

Reply:
Alright.  Making the interface optional and falling back to KUBECONFIG.